theano/sandbox/cuda/dnn.py

```diff
@@ -368,6 +368,36 @@ def connection_pattern(self, node):
         # not connected to desc
         return [[1], [1], [0]]
 
+    def infer_shape(self, node, shape):
+        b = shape[0][0]  # Number of inputs
+        h = shape[0][2]  # Height of input feature maps
+        w = shape[0][3]  # Width of input feature maps
+        nb = shape[1][0]  # Number of output feature maps
+        kh = shape[1][2]  # Height of each filter
+        kw = shape[1][3]  # Width of each filter
+        padh = 0
+        padw = 0
+        if (
+            not node.inputs[2].owner
+            or not isinstance(node.inputs[2].owner.op, GpuDnnConvDesc)
+        ):
+            raise theano.tensor.basic.ShareError("case not implemented and probably not needed")
```

Comment:
I fear that this case might get hit very often as the inputs to GpuDnnConvDesc will be mostly constants so it'll get contstant-folded.  To get around this you might need some GpuDnnShape Op or just use ctypes to forage through the C descriptor.